auto_round/quantizers/base.py

```diff
@@ -0,0 +1,1298 @@
+            low_gpu_mem_usage (bool, optional): Lower GPU memory mode. Defaults to False.
+            iters (int, optional): Optimization iterations. Defaults to 200.
+            seqlen (int, optional): Calibration sequence length. Defaults to 2048.
+            nsamples (int, optional): Number of calibration samples. Defaults to 128.
+            seed (int, optional): Random seed. Defaults to 42.
+            gradient_accumulate_steps (int, optional): Gradient accumulation steps. Defaults to 1.
+            data_type (str, optional): Weight data type string, e.g., "int". Defaults to "int".
+            act_bits (int, optional): Activation quantization bits. Defaults to 16.
+            act_group_size (int, optional): Activation group size. Defaults to None.
+            act_sym (bool, optional): Symmetric activation quantization. Defaults to None.
+            act_data_type (str, optional): Activation data type; inherits weight dtype if None and act_bits < 16.
+            act_dynamic (bool, optional): Dynamic activation quantization. Defaults to True.
+            enable_torch_compile (bool, optional): Enable torch.compile for quant blocks/layers. Defaults to False.
+            device_map (str | dict, optional): Device placement map. Defaults to None.
+            disable_opt_rtn (bool, optional): Disable RTN-mode optimization (iters=0). Defaults to False.
+            enable_alg_ext (bool, optional): Enable algorithm extension (primarily for INT2). Defaults to False.
+            **kwargs: Backward compatible options:
+                - enable_alg_ext, lr, lr_scheduler, sampler, not_use_best_mse, dynamic_max_gap,
+                  super_group_size, super_bits, scale_dtype ("fp16" etc.),
+                  nblocks, low_cpu_mem_usage, to_quant_block_names,
+                  enable_norm_bias_tuning, enable_quanted_input,
+                  disable_deterministic_algorithms, vlm, static_kv_dtype
+        Raises:
+            ValueError: If invalid device is provided or tokenizer is missing for non-str model with iters > 0.
+            RuntimeError: If model parameters are on meta device.
+        Example:
+            Layer-wise configuration structure:
+
+            >>> layer_config = {
+            ...     "layer1": {
+            ...         "data_type": "int",
+            ...         "bits": 4,
+            ...         "group_size": 128,
+            ...         "sym": True,
+            ...         "act_data_type": None,
+            ...         "act_bits": 16,
+            ...         "act_group_size": None,
+            ...         "act_sym": None,
+            ...     },
+            ...     # ...
+            ... }
+        """
+        self.scheme = None
+        self._parse_and_set_scheme(scheme, kwargs)
+
+        # Extra/legacy kwargs for backward compatibility
+        # Major version releases may pack them with extra configuration options
+        amp = kwargs.pop("amp", True)
```

Comment:
My initial thought is that quantizers should only include functions related to quantization, such as quantize_layer or quantize_block.